libpod/container_internal.go

```diff
@@ -492,11 +493,16 @@ func (c *Container) setupStorage(ctx context.Context) error {
 		containerInfo, containerInfoErr = c.runtime.storageService.CreateContainerStorage(ctx, c.runtime.imageContext, c.config.RootfsImageName, c.config.RootfsImageID, c.config.Name, c.config.ID, options)
 
 		if !generateName || !errors.Is(containerInfoErr, storage.ErrDuplicateName) {
+			errDuplicateName = true
 			break
 		}
 	}
 	if containerInfoErr != nil {
-		return fmt.Errorf("creating container storage: %w", containerInfoErr)
+		formatedErr := fmt.Errorf("creating container storage: %w", containerInfoErr)
+		if errDuplicateName {
+			formatedErr = fmt.Errorf("%w, or use --replace to instruct Podman to do so.", formatedErr)
```

Comment:
Now that I think of it I am not sure this advice makes sense in the general sense.
We can create containers in many different ways, for podman create/replace suggesting --replace makes sense but for API users this does not make sense at all. I don't think it is a good idea to hint to cli option inside of libpod where we have no idea how it was called. @vrothberg @mheon WDYT?

Reply:
Concur. We've softened on this rule considerably (we used to have absolutely no mentions of Podman inside Libpod) but we still try to keep the two as separate as possible.

Reply:
There are two parts to the issue, first we can figure out here and if it is external include that in the error message.
This could be done with `c.runtime.LookupContainer(c.config.Name)` if it returns an error we know it must be an external container so we could add this info the error message.

For the --replace suggestion we need to catch the error somewhere in cmd/podman... There is already a formatError() function that is called for all errors so we could use that.
https://github.com/containers/podman/blob/091d83d3c99a4322596e350f1b6431b7acd6e6df/cmd/podman/root.go#L627